test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl

```diff
@@ -1,73 +1,69 @@
+param newLineChar = 0x0A;
+param greaterThan = 0x3E;
 const pairs = "ATCGGCTAUAMKRYWWSSYRKMVBHDDHBVNN\n\n";
 var table : [1..128] uint(8);
 
-proc main(args: [] string) {
-  param newLineChar = 0x0A;
-  param greaterThan = 0x3E;
-  var inFile = openfd(0); // stdin
-  const fileLen = inFile.length();
-  var data : [1..fileLen] uint(8);
-  var r = inFile.reader(locking=false);
-
+proc main() {
   // initialize complement table
   for i in 0..#pairs.length by 2 {
-    table[pairs.buff[i]] = pairs.buff[i+1];      // uppercase
+    table[pairs.buff[i]] = pairs.buff[i+1];        // uppercase
     if pairs.buff[i] != newLineChar then
       table[pairs.buff[i] + 32] = pairs.buff[i+1]; // lowercase
   }
-
-  var numRead  : int;
+  
+  var data : string; 
   var idx = 1;
   var start = 0;
 
   // sync statements wait for all tasks inside them to complete
   sync {
-    while r.readline(data, numRead, idx) {
+    var data_in : string;
+    while readln(data_in) {
+      data += data_in;
 
       // Look for the start of a section, and if possible 
       // spawn a task to start work on the previous section.
-      if data[idx] == greaterThan {
-        if start == 0 then start = idx;
-        else {
+      if data.buff[idx] == greaterThan {
```

Comment:
While I like this idea, we'll need to convert between buff and string indices, either for this line or between main() and process(). I don't think there's a way to do it without making the code look funny.